### ISSUE TITLE
Cannot use data properties in script when using Vue.extend

### ISSUE THREAD

**Issue by Cy-Tek** [nuxt/typescript#82]
**Describe the bug**
When I have a template that is not using the class style of component, the parser complains about accessing data from either methods or computed properties. It does not complain when I convert the template to using `vue-property-decorator`

**To Reproduce**
1. Make a component with a script section like
    ```vue
    <script lang="ts">
    import Vue from 'vue'

    export default Vue.extend({
      name: 'Test',
      data() {
        return {
          str: 'Hello world'
        }
      },
      computed: {
        getStr(): string {
          return this.str
        }
      }
    })
    </script>
    ```
2. Run `nuxt-ts dev`
3. See error

**Expected behavior**
I should be able to access the properties on data from either the computed or methods properties without using `vue-property-decorator`

**Screenshots**
Example:
![image](https://user-images.githubusercontent.com/16470547/63535654-fe04f680-c4df-11e9-9c2e-01d6b7fec82b.png)

**Comment by kevinmarrec** [comment]
@Cy-Tek This issue may happen when you have a computed property that has implicit return type, making other return types buggy.

Minimal component following your code
![image](https://user-images.githubusercontent.com/25272043/63594131-ebf58780-c5b5-11e9-9404-44d6d5ff7215.png)
> Issue happenning cause return type of `authors` is implicit

Fixed by providing the correct return type :
![image](https://user-images.githubusercontent.com/25272043/63594230-370f9a80-c5b6-11e9-8377-4bc780d327cc.png)

I suspect that your component code contains somewhere an implicit typing that make the issue happens.

**Comment by kevinmarrec** [comment]
@Cy-Tek Can I close ?
